### PR TITLE
Mod1 pricing and basic styling

### PR DESCRIPTION
**Pull request by catligad**

### CODE REVIEW

File: server/index.js
Comment (Atr) [inline server/index.js:2]: If you install mysql using the terminal (ie, npm install --save mysql) then you likely shouldn't need the './'.  I would think just const mysql = require('mysql') would work.  Worth a try for consistency?

File: server/index.js
Comment (Atr) [inline server/index.js]: You should include a status code along with your response.

File: server/index.js
Comment (Atr) [inline server/index.js]: You should send back something to the client in the case of an error - an error status code at the very least.  Otherwise, in the case of an error the client's server request will hang.

File: server/index.js
Comment (Atr) [inline server/index.js]: I'm not quite sure, but something is telling me that you don't want the API path for your specific information to be the same as the API path for all information.  Did you clear this API path with Toby?
When we create our proxy server, our proxy server will receive a request for '/rooms/:listingID' and then based on that, it'll send specific requests to each of our module servers to serve up information specific to that module.  My gut is telling me the naming should be distinct.  Does that make sense? (you can take a look at my or Ivan's path names)

File: server/index.js
Comment (Atr) [inline server/index.js]: You have one table named using camelCase (userListing) and one using underscores (occupied_dates).  I think we're supposed to be consistent?

File: server/index.js
Comment (Atr) [inline server/index.js]: Using this syntax of populating the query with the listingID directly opens you up to SQL injection attacks.  Consider - but up to you! - using prepared queries.  Go here https://www.npmjs.com/package/mysql#preparing-queries and click prepared queries if you're interested.  Should only take a few minutes to implement.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: You should use this.setState inside this function to assign the proper id to the pathname state.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: I think you should delete this line and set the pathname state directly in your getPathname function.  See comment below.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx:84]: This implies you'll show nothing to your user if for some reason this.state.listingData doesn't exist.  Shouldn't you at least show a 'page not found' or some kind of simple error message?

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: Instead of using 'id', this should come directly from your state, right?  Ie this.state.pathname?

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: You shouldn't need to create another variable here because you're not modifying id.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: My read of this code is that if there's no id, then you're defaulting to an id of one.  I would think you should instead rely some kind of error in the case of a missing id.  Up to you if you want to change though.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: What if you get an error here?  I don't see a .catch.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx:72]: I like what you return here!  Very simple and clean : )

File: client/src/pricing_mod1/ModOne.jsx
Comment (Atr) [inline client/src/pricing_mod1/ModOne.jsx:26]: Nice use of defaultProps in case for some reason they aren't passed down!  You taught me something here, I may incorporate into my own code.

File: client/src/pricing_mod1/ModOne.jsx
Comment (Atr) [inline client/src/pricing_mod1/ModOne.jsx:41]: You used defaultProps in case no props were passed down.  But because 0 is a falsy value, you'll return 'null' if no props are passed down anyways.  So either using defaultProps is redundant if you want to return null in the case of no props passed down, or you should choose to return _something_ (not null) if price = 0.

File: client/src/pricing_mod1/PricePerNight.jsx
Comment (Atr) [inline client/src/pricing_mod1/PricePerNight.jsx:35]: See my comment on line 41 of ModOne.

File: client/src/pricing_mod1/Rating.jsx
Comment (Atr) [inline client/src/pricing_mod1/Rating.jsx:37]: See my comment on line 41 of ModOne.

**Review by Atr — COMMENTED** [review]
Hi Cat - looks good!  I'm submitting changes without approval because I think you should *consider* making some more changes before you merge.  Up to you what you decide to do, but at least take a look and then let me know so I can merge.  Re: your tests, awesome work that you figured out Jest and have so many!  I looked through more high-level to see that you were testing both styling and content and that you were covering each component.  I didn't take a deeper dive because I'm not up to speed on Jest yet, but let me know if you want me to.

File: server/index.js
Comment (catligad) [reply server/index.js:2]: You're right! I didn't even notice this. Thanks!

File: server/index.js
Comment (catligad) [reply server/index.js]: You're right!

File: server/index.js
Comment (catligad) [reply server/index.js]: Got it!

File: server/index.js
Comment (catligad) [reply server/index.js]: You're right I just didn't want to refactor lol but I'll get on that, thanks!

File: server/index.js
Comment (catligad) [reply server/index.js:2]: WAIT, just realized that this is a file lol I need it to be this way.

File: server/index.js
Comment (catligad) [reply server/index.js]: Thanks! Will do so! I didn't even think of this.

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: Got it! Had to promisify the fucntion :)

Comment (catligad) [reply client/src/index.jsx]: U R Correct

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: Got it!

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx:84]: It's for conditional rendering-- since I'm waiting for the initial state of the main app to be set before I render everything, I have to render null until I get the data that I need, otherwise there'll be no data to render it with.

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: Yup! Changed that thank you!

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: The linter was yelling at me :(

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: I didn't even need it I just realized because I had this code when I had errors a couple of days ago lol

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: U R Right

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx:72]: thanks!

File: client/src/pricing_mod1/ModOne.jsx
Comment (catligad) [reply client/src/pricing_mod1/ModOne.jsx:26]: The linter yells at you if you don't do it lol

File: client/src/pricing_mod1/ModOne.jsx
Comment (catligad) [reply client/src/pricing_mod1/ModOne.jsx:41]: The linter yells at you if you don't, so you have to. Either way, I need it to return null if no props are passed down.

File: server/index.js
Comment (Atr) [reply server/index.js:2]: OK!

File: client/src/pricing_mod1/ModOne.jsx
Comment (Atr) [reply client/src/pricing_mod1/ModOne.jsx:41]: Got it!  Makes sense.

File: client/src/index.jsx
Comment (Atr) [inline client/src/index.jsx]: I think you can use a relative path here (ie you don't need the http://127.0.0.1:3002 part) (?)

File: server/index.js
Comment (Atr) [inline server/index.js]: I like it, looks cleaner!

**Review by Atr — APPROVED** [review]

File: client/src/index.jsx
Comment (Atr) [reply client/src/index.jsx:84]: Makes sense!

File: client/src/index.jsx
Comment (catligad) [reply client/src/index.jsx]: You're right-- this would be better!

**Pull request merged**
